pkg/proxy/server.go

```diff
@@ -216,10 +223,12 @@ func NewServer(cfg ServerConfig) Server {
 	if err == nil {
 		s.fromPort, _ = strconv.Atoi(fromPort)
 	}
-	var toPort string
-	_, toPort, err = net.SplitHostPort(cfg.To.Host)
-	if err == nil {
-		s.toPort, _ = strconv.Atoi(toPort)
+	if !s.isHTTPProxy {
+		var toPort string
+		_, toPort, err = net.SplitHostPort(cfg.To.Host)
+		if err == nil {
+			s.toPort, _ = strconv.Atoi(toPort)
+		}
```

Comment:
nit: As `toPort` and `err` are only used in this scope (correct me if I'm wrong), this can be simplified to:

```suggestion		
		if _, toPort, err := net.SplitHostPort(cfg.To.Host); err == nil {
			s.toPort, _ = strconv.Atoi(toPort)
		}
```

I know you're not introducing this code, but it may be a good opportunity for a clean-up.

Reply:
You are right! Applied the suggestion! Thanks @ivanvc :)